File: implementations/csharp/ExecutionHints.cs
namespace Models
{
    /// <summary>
    /// User preferences
    /// </summary>
    public class ExecutionHints
    {
        public string PreferredModel {  get; set; }
        public Enums.ExecutionPriority Priority { get; set; } = Enums.ExecutionPriority.Balanced;
        public int MaxRetryCount { get; set; } = 3;
    }
}

File: implementations/vb/Loom.Core/Models/ExecutionHints.vb
Namespace Models
    ''' <summary>
    ''' User preferences
    ''' </summary>
    Public Class ExecutionHints
        Public Property PreferredModel As String
        Public Property Priority As Enums.ExecutionPriority = Enums.ExecutionPriority.Balanced
        Public Property MaxRetryCount As Integer = 3 ' error scenario

    End Class
End Namespace